components/camel-resteasy/src/main/java/org/apache/camel/component/resteasy/servlet/ResteasyCamelServlet.java

```
@@ -0,0 +1,359 @@
+     */
+    @SuppressWarnings("rawtypes")
+    @Override
+    public void init(ServletConfig servletConfig) throws ServletException {
+        super.init(servletConfig);
+
+        String name = servletConfig.getServletName();
+        if (httpRegistry == null) {
+            httpRegistry = DefaultHttpRegistry.getHttpRegistry(name);
+            ResteasyCamelServlet existing = httpRegistry.getCamelServlet(name);
+            if (existing != null) {
+                String msg = "Duplicate ServletName detected: " + name + ". Existing: " + existing + " This: " + this.toString()
+                        + ". Its advised to use unique ServletName per Camel application.";
+                // always log so people can see it easier
+                LOG.info(msg);
+            }
+            httpRegistry.register(this);
+        }
+
+
+        for (Map.Entry<String, HttpConsumer> entry : consumers.entrySet()) {
+            String proxyClasses = ((ResteasyComponent)getServletEndpoint(entry.getValue()).getComponent()).getProxyConsumersClasses();
+            if (proxyClasses != null) {
+                String[] classes = proxyClasses.split(",");
+                LOG.debug("Proxy classes defined in the component {}", Arrays.asList(classes));
+
+                for (String clazz : classes) {
+                    try {
+                        Class realClazz = Class.forName(clazz);
+                        // Create dynamic proxy class implementing interface
+                        InvocationHandler handler = new ResteasyInvocationHandler();
+                        Object  proxy = Proxy.newProxyInstance(realClazz.getClassLoader(), new Class[]{realClazz}, handler);
+
+                        // register new created proxy to the resteasy registry
+                        getDispatcher().getRegistry().addSingletonResource(proxy);
+                    } catch (ClassNotFoundException e) {
+                        e.printStackTrace();
```

Dont ignore exception, but log via logger or fail fast or something